io/native/src/main/scala/fs2/io/net/tls/S2nConfig.scala

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2013 Functional Streams for Scala
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2.io.net.tls
+
+import cats.effect.SyncIO
+import cats.effect.kernel.Resource
+import cats.effect.kernel.Sync
+import cats.effect.syntax.all._
+import cats.syntax.all._
+
+import scala.scalanative.unsafe._
+import scala.scalanative.unsigned._
+import scala.util.control.NonFatal
+
+import s2n._
+import s2nutil._
+
+/** An `s2n_config`.
+  *
+  * See [[https://github.com/aws/s2n-tls/ s2n-tls]] for detailed documentation.
+  */
+final class S2nConfig private (private[tls] val ptr: Ptr[s2n_config])
+
+object S2nConfig {
+
+  // we must call this *exactly* once, before doing anything else
+  // since you need an S2nConfig before you can do anything, this seems like the right place
+  s2n_init()
+
+  def builder: Builder = new BuilderImpl
+
+  sealed abstract class Builder {
+    def build[F[_]: Sync]: Resource[F, S2nConfig]
+
+    def withCertChainAndKeysToStore(certKeyPairs: List[CertChainAndKey]): Builder
+    def withPemsToTrustStore(pems: List[String]): Builder
+    def withWipedTrustStore: Builder
+    def withSendBufferSize(size: Int): Builder
+    def withVerifyHostCallback(cb: String => SyncIO[Boolean]): Builder
+    def withDisabledX509Verification: Builder
+    def withMaxCertChainDepth(maxDepth: Short): Builder
+    def withDHParams(dhparams: String): Builder
+    def withCipherPreferences(version: String): Builder
```

Comment:
s2n lets you specify your cipher preferences by selecting one of their pre-defined security policies.
https://github.com/aws/s2n-tls/blob/main/docs/USAGE-GUIDE.md#security-policies

Note that the default policy (aka "default") does not use TLS1.3; you have to opt-in with "default_tls13".